gobbler-server.rkt

```diff
@@ -222,9 +222,24 @@ waiting list.
 
 ;; GobblerUniverse iworld? -> GobblerBundle
 ;; Remove player from the game
-(define (drop-world uni world)
+(define (drop-world! uni world)
+  (set-game-queue! uni (drop-queued (game-queue uni) world))
+  (when (ready? uni)
+    (set-ready-players! uni (drop-player (ready-players uni) world)))
+  
   uni)
 
+;; [Listof iworld?] iworld? -> [Listof iworld?]
+;; Drop the iworld from the game queue
+(define (drop-queued queue world)
+  (remove world queue))
+
+;; [Listof player?] iworld? -> [Listof player?]
+;; Drop the player with the iworld from the game
+(define (drop-player players world)
+  (filter (λ (player) (not (iworld=? world (player-iworld player))))
+          players))
```

Comment:
This one can't be implemented using `remove` because `world` won't be equal to any of the players. It needs to be compared to each player's `iworld`.